### DISCUSSION TITLE
Sunricher ZG9101SAC-HP-Switch-B

### DISCUSSION THREAD

**Discussion by oscarbeg** [Koenkk/zigbee2mqtt#23449] [Devices]
Hi there,
I am trying to pair this Sunricher ZG9101SAC-HP-Switch-B device in HA but it is detected as the previous version Sunricher ZG9101SAC-HP-Switch. The new version -B ZigBee clusters supported by the device are as follows:
Input Cluster
- 0x000000: Basic - 0x0003: Identify - 0x0004: Groups - 0x0005: Scenes - 0x0006: Power On/Off
- 0x0702: Simple measurement - 0x0b04: Electrical measurement - 0x0b05: Diagnostics
Output Groups
- 0x0019: OTA
How can we integrate this new device so that it is recognised correctly?
Or if it recognises it as the previous version, how can I add the new consumption and electrical measurement groups?
Thank you very much.

**Comment by pieterh** [comment]
Hi Oscar, I'm looking to buy this device so I will be able to control a device that has a high consumption.
You seem to be able to control it even though it is recognised as an other version.
Do you now if the energy property is read properly?

**Comment by andrei-lazarov** [comment]
Should be fixed in next Z2M release:
- https://github.com/Koenkk/zigbee-herdsman-converters/pull/11396
